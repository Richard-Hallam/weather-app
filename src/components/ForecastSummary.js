import React from "react";
import forecast from "../data/forecast.json";
import iconData from "../data/iconData.json";


function simplifyWeatherCode(weatherCode){
    let simplifiedCode = 0;
    if (weatherCode < 300 && weatherCode >= 200){
        simplifiedCode = 200;
    } else if (weatherCode < 400 && weatherCode >= 300){
        simplifiedCode = 300;
    }else if (weatherCode < 600 && weatherCode >= 500){
        simplifiedCode = 500;
    }else if (weatherCode < 700 && weatherCode >= 600){
        simplifiedCode = 600;
    }else if (weatherCode < 800 && weatherCode >= 700){
        simplifiedCode = 700;
    }else if (weatherCode < 900 && weatherCode >= 800){
        simplifiedCode = 800;
    }
    return simplifiedCode;
}


function ForecastSummary(props){
    const {
        date,
        description,
        icon,
        temperature,
    } = props;

    
    const formattedDate = new Date(date).toDateString();
    

    return<div className="forecast-summary" data-testid="forecast-summary">
        <div className="forecast-summary_date">
            {formattedDate}
        </div>
        <div className="forecast-summary_icon" data-testid="forecast-icon">
            <img src = {iconData[simplifyWeatherCode(icon)]} alt='Weather Icon'/>
        </div>
        <div className="forecast-summary_temperature">
            {temperature.max}
        </div>
        <div className="forecast-summary_description">
            {description}
        </div>
    </div>;
};



export default ForecastSummary;