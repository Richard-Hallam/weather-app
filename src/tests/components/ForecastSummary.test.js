import React from 'react';
import { getByText, render } from '@testing-library/react';
import ForecastSummary from '../../components/ForecastSummary';

describe("LocationDetails", () => {
    const validProps ={
        date:1111111,
        description:"Stub description",
        icon: "stubIcon",
        temperature:{
            min: 12,
            max: 22,
        },
    };

    it('renders the forecast summary', () => {
        const { getByText, getByTestId} = render(
            <ForecastSummary
            date={validProps.date}
            description={validProps.description}
            icon={validProps.icon}
            temperature={validProps.temperature}
        />
        );

        expect(getByText("1111111")).toHaveAttribute("class", "forecast-summary_date");
        expect(getByText("Stub description")).toHaveAttribute("class", "forecast-summary_description");
        expect(getByText("stubIcon")).toHaveAttribute("class", "forecast-summary_icon");
        expect(getByText("22")).toHaveAttribute("class", "forecast-summary_temperature");
    });
});