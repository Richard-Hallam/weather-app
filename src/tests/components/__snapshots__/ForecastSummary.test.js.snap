// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
    data-testid="forecast-summary"
  >
    <div
      class="forecast-summary_date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-summary_icon"
      data-testid="forecast-icon"
    >
      <img
        alt="Weather Icon"
        src="https://openweathermap.org/img/wn/01d@2x.png"
      />
    </div>
    <div
      class="forecast-summary_temperature"
    >
      22Â°C
    </div>
    <div
      class="forecast-summary_description"
    >
      Stub description
    </div>
    <button
      type="button"
    >
      More details
    </button>
  </div>
</DocumentFragment>
`;
